camayoc/tests/qpc/api/v1/credentials/test_network_creds.py

```diff
@@ -36,18 +37,22 @@ def test_update_password_to_sshkeyfile(shared_client, cleanup, isolated_filesyst
         3) Confirm network credential has been updated.
     :expectedresults: The network credential is updated.
     """
-    cred = Credential(cred_type="network", client=shared_client, password=uuid4())
-    cred.create()
-    # add the id to the list to destroy after the test is done
-    cleanup.append(cred)
+    cred = next(
+        data_provider.credentials.new(
+            match_criteria={"type": "network", "password": Table.is_not_null()},
+            data_only=False,
+        )
+    )
```

Comment:
This is more of an exercise. We never run scan with that credential, so we don't need a valid password. However, it shows how we can save typing `cred.create()` and `cleanup.append(cred)` all over again - both can be handled by DataProvider.